rclcpp_lifecycle/test/test_lifecycle_node.cpp

```diff
@@ -34,6 +34,85 @@
 using lifecycle_msgs::msg::State;
 using lifecycle_msgs::msg::Transition;
 
+static
+bool wait_for_event(
+  std::shared_ptr<rclcpp_lifecycle::LifecycleNode> node,
+  std::function<bool()> predicate,
+  std::chrono::nanoseconds timeout,
+  std::chrono::nanoseconds sleep_period)
+{
+  auto start = std::chrono::steady_clock::now();
+  std::chrono::microseconds time_slept(0);
+
+  bool predicate_result;
+  while (!(predicate_result = predicate()) &&
+    time_slept < std::chrono::duration_cast<std::chrono::microseconds>(timeout))
+  {
+    rclcpp::Event::SharedPtr graph_event = node->get_graph_event();
+    node->wait_for_graph_change(graph_event, sleep_period);
+    time_slept = std::chrono::duration_cast<std::chrono::microseconds>(
+      std::chrono::steady_clock::now() - start);
+  }
+  return predicate_result;
+}
+
+static
+bool wait_for_topic(
+  std::shared_ptr<rclcpp_lifecycle::LifecycleNode> node,
+  const std::string & topic,
+  std::chrono::nanoseconds timeout = std::chrono::seconds(3),
+  std::chrono::nanoseconds sleep_period = std::chrono::milliseconds(100))
```

Comment:
while these timeouts usually work, they are still prone to flakiness IMO. That is, we might increase or decrease these over time. It's a very minor nit, but you could pull these out into constants in case we have to deal with them later on again.

Reply:
I opted for parameters to the function so they could be adapted individually if necessary, though it's more likely we'd just update these defaults if we needed to bump the timeout. I think something else is very wrong if a node can't query it's own topics/services within 3 seconds :fearful:

Reply:
I guess a resource constrained machine could cause us to exceed the timeout.

Reply:
See 71c392a

Reply:
I think you have it in line 103/104 again.